libevmasm/Assembly.cpp

```diff
@@ -227,16 +239,19 @@ AssemblyItem Assembly::createAssemblyItemFromJSON(Json const& _json, std::vector
 		else if (name == "PUSH [tag]")
 		{
 			requireValueDefinedForInstruction(name, value);
+			requireValueInRange(name, value, 0, std::numeric_limits<unsigned>::max());
```

Comment:
I'm not 100% sure `unsigned` is the right type here. We convert the `value` to this type in many places, including when generating the bytecode so it seems to me that higher values cannot work, but there's one bit of code that seems to contradict it: https://github.com/ethereum/solidity/blob/1cce974400d80e4002d4fd37ab432c79260cca8b/libevmasm/AssemblyItem.cpp#L52-L70

`toSubAssemblyTag()` and `splitForeignPushTag()` apparently expect the value to be `u256` with the tag stored in the lower 64 bits. Not sure how to square that with the rest of the code. Is that structure used only temporarily, never reaching the assembling stage?